lib/ansible/plugins/inventory/nmap.py

```diff
@@ -72,18 +70,35 @@ def __init__(self):
         self._nmap = None
         super(InventoryModule, self).__init__()
 
-    def verify_file(self, path):
+    def _populate(self, hosts):
+        # Use constructed if applicable
+        strict = self.get_option('strict')
+
+        for host in hosts:
+            hostname = host['name']
+            self.inventory.add_host(hostname)
+            for var, value in host.items():
+                self.inventory.set_variable(hostname, var, value)
+
+            # Composed variables
+            self._set_composite_vars(self.get_option('compose'), host, hostname, strict=strict)
+
+            # Complex groups based on jinja2 conditionals, hosts that meet the conditional are added to group
+            self._add_host_to_composed_groups(self.get_option('groups'), host, hostname, strict=strict)
 
+            # Create groups based on variable values and add the corresponding hosts to it
+            self._add_host_to_keyed_groups(self.get_option('keyed_groups'), host, hostname, strict=strict)
+
+    def verify_file(self, path):
+        ''' return true/false if this is possibly a valid file for this plugin to consume '''
         valid = False
         if super(InventoryModule, self).verify_file(path):
-            file_name, ext = os.path.splitext(path)
-
-            if not ext or ext in C.YAML_FILENAME_EXTENSIONS:
+            # base class verifies that file exists and is readable by current user
+            if path.endswith(('nmap.yaml', 'nmap.yml')):
```

Comment:
Something like:
```suggestion
            if path.endswith(['nmap' + e for e in C.YAML_FILENAME_EXTENSIONS]):
```
would keep a users pref for yaml extensions.

Reply:
I do agree that this is better, with the doc for configuration settings talking about inventory using that var. Something I find weird though is that the doc for developing inventory plugin is not honoring that var in the virtualbox example